docs/broadcast-guide.md

```diff
@@ -0,0 +1,89 @@
+
+Example:
+
+```csharp
+string v = "Variable to be broadcasted";
+Broadcast<string> bv = SparkContext.Broadcast(v);
+
+// Using the broadcast variable in a UDF:
+Func<Column, Column> udf = Udf<string, string>(
+	str => $"{str}: {bv.Value()}");
+```
+
+The type of broadcast variable is captured by using Generics in C# as can be seen in the above example.
+
+### Deleting broadcast variables
+
+The broadcast variable can be deleted from all executors by calling the `Destroy()` function on it.
+
+```csharp
+// Destroying the broadcast variable bv:
+bv.Destroy();
+```
+
+> Note: `Destroy` deletes all data and metadata related to the broadcast variable. Use this with caution; once a broadcast variable has been destroyed, it cannot be used again.
+
+#### Caveat of using Destroy
+
+One important thing to keep in mind while using Broadcast variables in UDFs is to limit the scope of the variable to only the UDF that is referencing it. The [guide to using UDFs](udf-guide.md) describes this phenomenon in detail. This is especially crucial when calling `Destroy` on the broadcast variable, as if the broadcast variable that has been destroyed is visible to/accessible from other UDFs, it gets picked up for serialization by all those UDFs even if it is not being referenced by them; and throws an error as .NET for Apache Spark is not able to serialize the destroyed broadcast variable.
+
+Example to demonstrate:
+
+```csharp
+string v = "Variable to be broadcasted";
+Broadcast<string> bv = SparkContext.Broadcast(v);
+
+// Using the broadcast variable in a UDF:
+Func<Column, Column> udf1 = Udf<string, string>(
+	str => $"{str}: {bv.Value()}");
+
+// Destroying bv
+bv.Destroy();
+
+// Calling udf1 after destroying bv throws the following expected exception:
+// org.apache.spark.SparkException: Attempted to use Broadcast(0) after it was destroyed
+df.Select(udf1(df["_1"])).Show();
+
+// Different UDF udf2 that is not referencing bv
+Func<Column, Column> udf2 = Udf<string, string>(
+	str => $"{str}: not referencing broadcast variable");
+
+// Calling udf2 throws the following (unexpected) exception:
+// [Error] [JvmBridge] org.apache.spark.SparkException: Task not serializable
+df.Select(udf2(df["_1"])).Show();
+```
+
+The recommended way of implementing above desired behavior:
+
+```csharp
+string v = "Variable to be broadcasted";
+// Restricting the visibility of bv to only the UDF referencing it
+{
+	Broadcast<string> bv = SparkContext.Broadcast(v);
+
+	// Using the broadcast variable in a UDF:
+	Func<Column, Column> udf1 = Udf<string, string>(
+		str => $"{str}: {bv.Value()}");
+
+	// Destroying bv
+	bv.Destroy();
+}
+
+// Different UDF udf2 that is not referencing bv
+Func<Column, Column> udf2 = Udf<string, string>(
+	str => $"{str}: not referencing broadcast variable");
+
+// Calling udf2 works fine as expected
+df.Select(udf2(df["_1"])).Show();
+```
```

Comment:
This guide looks great- I didn't know what broadcast variables were, and now I feel like I actually understand them :) Do you want to end this doc with a little conclusion/wrap up section, or a link to a guide that would make sense for users to read over next? i.e. maybe mention the udf guide again, or link to a coding example or article that uses broadcast.

Reply:
Yes I was thinking about that, I like all the suggestions you gave, I can conclude with a summary and then link for an example to use broadcast variables, what do you think?

Reply:
That sounds great! :)